sdk/core/Azure.Core/src/Shared/RetriableStream.cs

```diff
@@ -66,14 +65,10 @@ private class RetriableStreamImpl : Stream
 
             public RetriableStreamImpl(Stream initialStream, Func<long, Stream> streamFactory, Func<long, ValueTask<Stream>> asyncStreamFactory, ResponseClassifier responseClassifier, int maxRetries)
             {
-                try
+                if (initialStream.CanSeek)
```

Comment:
do we care about misbehaving streams? (i.e. should we leave catch just in case like we did with CanTimeout?)

Reply:
Added